### ISSUE TITLE
iOS top task bar is white background and white icons when flutter appbar is white.

### ISSUE THREAD

**Issue by zhenqiu1101** [flutter/flutter#12173]
If I create a scaffold like below,

```dart
new Scaffold(
        appBar: new AppBar(
          leading: new BackButton(color: Colors.grey[600]),
          title: _buildSearchBar,
          backgroundColor: Colors.white,
        ),
        backgroundColor: Colors.white,
        body: body);
```

In Android, the search bar is white and the top task bar (a thin bar at very top showing time, battery and icons) is grey.

In iOS, the search bar is white but and top task bar is white too. It masks all the white icons and timestamp so we can't see anything.

**Comment by mehmetf** [comment]
@Hixie What do you think about this? I would imagine having a search bar in white is quite common.

**Comment by Hixie** [comment]
(Side note: when changing this kind of thing I would recommend just wrapping the AppBar in an appropriate Theme so that you don't have to individually control each component.)

This is a dupe of https://github.com/flutter/flutter/issues/4630; in the meantime you can work around it by manually calling `SystemChrome.setSystemUIOverlayStyle()`.

**Comment by jiajiabingcheng** [comment]
Hi Hixie,

I tried `SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.light);`, but it doesn't work. That's because this method also get called inside the appbar build method.

The theme of the status bar is always the same as the theme of the AppBar. So it works with manually setting the app bar brightness to Brightness.light when build a single Scaffold widget with the current primaryColorBrightness == Brightness.dark. In this case the system icons (carrier, battery and etc.) are in black. 

But when this Scaffold widget(2) is in the MaterialPageRoute builder and pushed from another Scaffold widget(1) with Theme.primaryColorBrightness == Brightness.dark, it doesn't work for the first build of wiget (2). It works when widget(2) get a rebuild. 

I just filed a bug related to this issue(https://github.com/flutter/flutter/issues/12960) :]
